Content.Shared/_RMC14/Entrenching/EntrenchingToolSystem.cs

```diff
@@ -48,10 +50,62 @@ private void OnAfterInteract(Entity<EntrenchingToolComponent> tool, ref AfterInt
         if (!args.CanReach)
             return;
 
+        if (HasComp<BarricadeSandbagComponent>(args.Target))
+        {
+            DismantleSandbagBaricade(tool, ref args);
+            args.Handled = true;
+            return;
+        }
+
         StartDigging(tool, args.User, args.ClickLocation);
         args.Handled = true;
     }
 
+    private void DismantleSandbagBaricade(Entity<EntrenchingToolComponent> tool, ref AfterInteractEvent args)
+    {
+        if (TryComp(tool, out ItemToggleComponent? toggle) && !toggle.Activated)
+            return;
+
+        _popup.PopupClient(Loc.GetString("cm-entrenching-dismantle"), args.User, args.User);
+
+        var ev = new SandbagDismantleDoAfterEvent(GetNetCoordinates(args.ClickLocation));
+        var doAfter = new DoAfterArgs(EntityManager, args.User, tool.Comp.DigDelay, ev, tool, args.Target, tool)
+        {
+            BreakOnMove = true,
+        };
+        _doAfter.TryStartDoAfter(doAfter);
+    }
+
+    private void OnSandbagDismantleDoAfter(Entity<EntrenchingToolComponent> tool, ref SandbagDismantleDoAfterEvent args)
+    {
+        if (args.Cancelled || args.Handled)
+            return;
+
+        args.Handled = true;
+
+        if (_net.IsServer)
+        {
+            var full = Spawn("CMSandbagFull", EntityManager.GetCoordinates(args.Coordinates));
```

Comment:
CMSandbagFull here should be a datafield on the barricade of type EntProtoId, if nothing else so that it automatically gets validated to exist, but also in case it ever needs to be changed from YAML